tests/contracts/conftest.py

```diff
@@ -0,0 +1,81 @@
+import os
+import pytest
+import eth_tester
+from eth_tester import (
+    EthereumTester,
+    PyEVMBackend
+)
+from web3.providers.eth_tester import (
+    EthereumTesterProvider,
+)
+from web3 import (
+    Web3,
+)
+from web3.contract import (
+    ConciseContract,
+)
+
+from vyper import (
+    compiler,
+    utils as vyper_utils,
+)
+
+REGISTRATION_DEPOSIT = 32
+
+
+def get_dirs(path):
+    own_dir = os.path.dirname(os.path.realpath(__file__))
+    abs_contract_path = os.path.realpath(
+        os.path.join(own_dir, '..', '..', 'contracts'))
+    sub_dirs = [x[0] for x in os.walk(abs_contract_path)]
+    extra_args = ' '.join(['{}={}'.format(d.split('/')[-1], d)
+                           for d in sub_dirs])
+    path = '{}/{}'.format(abs_contract_path, path)
+    return path, extra_args
+
+
+@pytest.fixture
+def registration_code():
+    with open(get_dirs('validator_registration.v.py')[0]) as f:
+        return f.read()
+
+
+@pytest.fixture
+def base_tester():
+    return EthereumTester(PyEVMBackend())
+
+
+@pytest.fixture
+def a0(base_tester):
+    return base_tester.get_accounts()[0]
+
+
+@pytest.fixture
+def w3(base_tester):
+    web3 = Web3(EthereumTesterProvider(base_tester))
+    return web3
+
+
+@pytest.fixture
+def tester(w3, base_tester, registration_code):
+    contract_bytecode = compiler.compile(registration_code)
+    contract_abi = compiler.mk_full_signature(registration_code)
+    registration = w3.eth.contract(
+        abi=contract_abi, bytecode=contract_bytecode)
+    tx_hash = registration.constructor().transact()
+    tx_receipt = w3.eth.waitForTransactionReceipt(tx_hash)
+    registration_deployed = w3.eth.contract(
+        address=tx_receipt.contractAddress,
+        abi=contract_abi
+    )
+    return base_tester, registration_deployed
```

Comment:
I would decouple `tester` and `registration` fixtures for clarity when "importing" them into a test.

```python
@pytest.fixture
def base_tester():
    return EthereumTester(PyEVMBackend())

@pytest.fixture
def registration_contract(w3, base_tester, registration_code):
    contract_bytecode = compiler.compile(registration_code)
    contract_abi = compiler.mk_full_signature(registration_code)
    registration = w3.eth.contract(
        abi=contract_abi, bytecode=contract_bytecode)
    tx_hash = registration.constructor().transact()
    tx_receipt = w3.eth.waitForTransactionReceipt(tx_hash)
    registration_deployed = w3.eth.contract(
        address=tx_receipt.contractAddress,
        abi=contract_abi
    )
    return registration_deployed
```

In the casper contract, we did the big `tester` fixture because we had a lot to compile and deploy and really cared where that casper contract and all the helper contracts ended up. In this case, we can just deploy the contract where ever and interact with it. Most of the tests would even need `tester` fixture, they'll just need `registration_contract`.

Reply:
We also had `base_tester` different from `tester` so we could test a blank chain or a casper-enabled chain depending on our needs